src/rules.rs

```diff
@@ -4,238 +4,233 @@ use serde::{Deserialize, Serialize};
+pub struct Rules {
+    // Invert all matchings
+    #[serde(default)]
+    pub invert: bool,
+
+    #[serde(default)]
+    pub subjects: SubjectRules,
+
+    #[serde(default)]
+    pub objects: ObjectRules,
 }
 
-pub fn match_type_rule_named_node(
-    is_subject: bool,
-    n: &NamedNode,
-    mask: TripleMask,
+/// Check all parts of the triple against rules.
+pub fn match_rules(
+    triple: &Triple,
     rules: &Rules,
     type_map: &HashMap<String, String>,
 ) -> TripleMask {
-    let iri_type = if let Some(v) = type_map.get(&n.iri) {
-        v
-    } else {
-        // Not in the type map.
-        return mask;
-    };
+    let mut mask =
+        match_subject_rules(triple, rules, type_map) | match_object_rules(triple, rules, type_map);
 
-    if !rules.replace_uri_of_nodes_with_type.contains(iri_type) {
-        // Not in the rules.
-        return mask;
+    if rules.invert {
+        mask = mask.invert();
     }
 
-    return if is_subject {
-        mask | TripleMask::SUBJECT
-    } else {
-        mask | TripleMask::OBJECT
-    };
+    return mask;
 }
 
-pub fn match_type_rule_subject(
-    subject: &Subject,
-    mask: TripleMask,
-    type_map: &HashMap<String, String>,
+/// Checks subject and object against subject-rules.
+pub fn match_subject_rules(
+    triple: &Triple,
     rules: &Rules,
+    type_map: &HashMap<String, String>,
 ) -> TripleMask {
-    match subject {
-        Subject::NamedNode(n) => {
-            return mask | match_type_rule_named_node(true, n, mask, rules, type_map);
-        }
-        Subject::BlankNode(_) => return mask,
-    }
+    let pseudo_subject = match &triple.subject {
+        Subject::NamedNode(n) => match_type(&n.iri, rules, type_map),
+        _ => false,
+    };
+    let pseudo_object = match &triple.object {
+        Term::NamedNode(n) => match_type(&n.iri, rules, type_map),
+        _ => false,
+    };
+
+    let mut mask = TripleMask::default();
+    if pseudo_subject {
+        mask |= TripleMask::SUBJECT;
+    };
+    if pseudo_object {
+        mask |= TripleMask::OBJECT;
+    };
+
+    return mask;
 }
 
-pub fn match_type_rule_object(
-    object: &Term,
-    mask: TripleMask,
-    type_map: &HashMap<String, String>,
+/// Checks triple against object rules
+pub fn match_object_rules(
+    triple: &Triple,
     rules: &Rules,
+    type_map: &HashMap<String, String>,
 ) -> TripleMask {
-    match object {
+    let pseudo_object = match &triple.object {
         Term::NamedNode(n) => {
-            return mask | match_type_rule_named_node(false, n, mask, rules, type_map);
+            if match_predicate(&n.iri, rules) {
+                true
+            } else {
+                match_type_predicate(&n.iri, &triple.predicate.iri, type_map, rules)
+            }
         }
-        _ => return mask,
-    }
-}
-
-pub fn match_predicate_rule(predicate: &NamedNode, mask: TripleMask, rules: &Rules) -> TripleMask {
-    let NamedNode { iri: i } = predicate;
+        _ => false,
```

Comment:
issue(bug): sneaky missing `Literal` type. When running `tripsu` on the example data and config, "accessCode" values are not pseudonymized. The rules all work as expected and after a bit of investigation I landed on this line. I think that for best practices, it's better to be explicit on the types when using `match` and avoid using `_`.

Reply:
i.e. the fix in [544fceb](https://github.com/sdsc-ordes/tripsu/pull/47/commits/544fceb54f507b42c437c923069ee57cefac7eb9), but I'll add a test in second_pass as you suggested